benchmarks/pipe.py

```diff
@@ -497,64 +401,70 @@ def generate_balance(num_devices, num_layers):
     return balance
 
 
-def make_model_and_data(args, device, new_data: bool = True):
+def make_model_and_data(args, new_data: bool = True, config=None):
     device = torch.device("cuda") if torch.cuda.is_available() else torch.device("cpu")
-    if new_data:
-        vocab_size = 10000
-        model, criterion, optimizer, scaler = make_model(args, device, vocab_size)
+    if args.use_synthetic_data:
+        model, optimizer = make_model(args, device, config)
         lm_dataset = BenchmarkLMDataset()
         lm_dataloader = DataLoader(
             lm_dataset, batch_size=args.batch_size, shuffle=True, num_workers=0, collate_fn=collate_sentences_lm
         )
-        return {
-            "model": model,
-            "criterion": criterion,
-            "optimizer": optimizer,
-            "data": lm_dataloader,
-            "vocab_size": vocab_size,
-        }
+        return {"model": model, "optimizer": optimizer, "data": lm_dataloader}
     else:
-        data = get_data(device)
-        ntokens, train_data, val_data, test_data = data
-        model, criterion, optimizer, scaler = make_model(args, device, ntokens)
+        data = datasets.get_wikitext2_data(device)
+        ntokens, _, _, _ = data
+        config["vocab_size"] = ntokens
+        model, optimizer = make_model(args, device, ntokens)
         return {
             "model": model,
-            "criterion": criterion,
             "optimizer": optimizer,
             "data": data,
         }
 
 
-def bench_single_process(args):
+def create_benchmark_config(model_name):
```

Comment:
docstring, esp. the return type should be documented.

Reply:
Done.